src/rules/pydocstyle/rules/non_imperative_mood.rs

```diff
@@ -14,6 +15,38 @@ static MOOD: Lazy<Mood> = Lazy::new(Mood::new);
 
 /// D401
 pub fn non_imperative_mood(checker: &mut Checker, docstring: &Docstring) {
+    let (
+        DefinitionKind::Function(parent)
+        | DefinitionKind::NestedFunction(parent)
+        | DefinitionKind::Method(parent)
+    ) = &docstring.kind else {
+        return;
+    };
+    match &parent.node {
+        StmtKind::FunctionDef {
+            name,
+            decorator_list,
+            ..
+        }
+        | StmtKind::AsyncFunctionDef {
+            name,
+            decorator_list,
+            ..
+        } => {
+            if name.starts_with("test") || name.eq("runTest") {
+                return;
+            }
+            for expr in decorator_list.iter() {
+                if let ExprKind::Name { id, .. } = &expr.node {
+                    if id.eq("property") {
+                        return;
```

Comment:
It looks like Pydocstyle defaults to accepting `property,cached_property,functools.cached_property`.

Maybe just for consistency with this "kind" of check, let's add a method to `src/visibilty.rs` like:

```rs
pub fn is_property(checker: &Checker, decorator_list: &[Expr]) -> bool {
    decorator_list.iter().any(|expr| {
        checker.resolve_call_path(expr).map_or(false, |call_path| {
            call_path.as_slice() == ["", "property"] || call_path.as_slice() == ["functools", "cached_property"]
        })
    })
}
```

That will check the additional decorator _and_ ensure that it only triggers when `functools` is imported _and_ track aliases _and_ track overrides :)

Reply:
I actually implemented `is_property` in `visibility.rs` in my first cut of this patch, but then decided to move it back, "inlined" at the call site. I felt `@property` isn't actually related to visibility in the broad sense the other similar helpers there are, and thus it seemed misplaced.

But no strong opinions really, could you confirm if you'd still rather see it there and I'll refactor.

In any case, will revise this tomorrow, addressing the other stuff. Thanks!

Reply:
Yeah I think that's a totally reasonable instinct, but I'd still rather see it in there. There are some other helpers in that file that aren't _really_ related to visibility (like `is_override`), and I'd like to have them logically grouped for now.

Reply:
Sure, done in f38b51a29dc243060f77c6d90e6e453242f1ef7e.